File: monetran-lrw/backend/diem_utils/types/currencies.py
# Copyright (c) The Diem Core Contributors
# SPDX-License-Identifier: Apache-2.0

import typing
from enum import Enum


class DiemCurrency(str, Enum):
    XUS = "XUS"
    USDC = "USDC"


class WyreCurrency(str, Enum):
    USDC = "USDC"
    USD = "USD"
    EUR = "EUR"
    GBP = "GBP"
    CHF = "CHF"
    CAD = "CAD"
    AUD = "AUD"
    NZD = "NZD"
    JPY = "JPY"
    MXN = "MXN"


class FiatCurrency(str, Enum):
    USD = "USD"
    EUR = "EUR"
    GBP = "GBP"
    CHF = "CHF"
    CAD = "CAD"
    AUD = "AUD"
    NZD = "NZD"
    JPY = "JPY"
    MXN = "MXN"


Currencies = typing.Union[FiatCurrency, DiemCurrency]

DEFAULT_DIEM_CURRENCY = DiemCurrency.XUS.value

File: monetran-lrw/mobile/src/interfaces/currencies.ts
// Copyright (c) The Diem Core Contributors
// SPDX-License-Identifier: Apache-2.0

export type DiemCurrency = "XDM" | "XUS" | "USDC" | "Coin2";

export type FiatCurrency = "USD" | "EUR" | "GBP" | "CHF" | "CAD" | "AUD" | "NZD" | "JPY" | "MXN";

export type DiemCurrenciesSettings = {
  [key in DiemCurrency]: {
    name: string;
    symbol: DiemCurrency;
    sign: string;
  };
};

export type FiatCurrenciesSettings = {
  [key in FiatCurrency]: {
    symbol: FiatCurrency;
    sign: string;
  };
};

export type Rates = {
  [key in FiatCurrency | DiemCurrency]: {
    [key in FiatCurrency | DiemCurrency]: number;
  };
};
